airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/source.py

```diff
@@ -22,69 +22,105 @@
+        role = boto3_client.assume_role(RoleArn=config.role_arn, RoleSessionName="guid")
+        role_creds = role["Credentials"]
+        aws_signature = AWSSignature(
+            service="execute-api",
+            aws_access_key_id=role_creds.get("AccessKeyId"),
+            aws_secret_access_key=role_creds.get("SecretAccessKey"),
+            aws_session_token=role_creds.get("SessionToken"),
+            region=self.region,
+        )
+        auth = AWSAuthenticator(
+            token_refresh_endpoint="https://api.amazon.com/auth/o2/token",
+            client_secret=config.lwa_client_secret,
+            client_id=config.lwa_app_id,
+            refresh_token=config.refresh_token,
+            host=self.endpoint.replace("https://", ""),
+        )
+        stream_kwargs = {
+            "url_base": self.endpoint,
+            "authenticator": auth,
+            "aws_signature": aws_signature,
+            "replication_start_date": config.replication_start_date,
+        }
+        return stream_kwargs
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
         try:
-            client.check_connection()
-            return AirbyteConnectionStatus(status=Status.SUCCEEDED)
-        except Exception as e:
-            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {str(e)}")
-
-    def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
-        client = self._get_client(config)
-
-        return AirbyteCatalog(streams=client.get_streams())
-
-    def read(
-        self, logger: AirbyteLogger, config: Mapping[str, Any], catalog: ConfiguredAirbyteCatalog, state: MutableMapping[str, Any] = None
-    ) -> Generator[AirbyteMessage, None, None]:
-        client = self._get_client(config)
-
-        logger.info("Starting syncing Amazon Seller API")
-        for configured_stream in catalog.streams:
-            yield from self._read_record(logger=logger, client=client, configured_stream=configured_stream, state=state)
-
-        logger.info("Finished syncing Amazon Seller API")
-
-    @staticmethod
-    def _read_record(
-        logger: AirbyteLogger, client: BaseClient, configured_stream: ConfiguredAirbyteStream, state: MutableMapping[str, Any]
-    ) -> Generator[AirbyteMessage, None, None]:
-        stream_name = configured_stream.stream.name
-        is_report = client._amazon_client.is_report(stream_name)
-
-        if configured_stream.sync_mode == SyncMode.full_refresh:
-            state.pop(stream_name, None)
-
-        if is_report:
-            yield from client.read_reports(logger, stream_name, state)
-        else:
-            yield from client.read_stream(logger, stream_name, state)
+            config = ConnectorConfig.parse_obj(config)  # FIXME: this will be not need after we fix CDK
+            stream_kwargs = self._get_stream_kwargs(config)
```

Comment:
This would work fine. 
In case if it's type error, we'll get smth like `Unable to connect to Amazon Seller API with the provided credentials - ValidationError(model=ConnectorConfig, errors=[{loc: (aws_environment,), msg: "value is not a valid enumeration member; permitted: PRODUCTION, SANDBOX", type: type_error.enum, ctx: {enum_values: [<AWSEnvironment.PRODUCTION: PRODUCTION>, <AWSEnvironment.SANDBOX: SANDBOX>]}}])`. Which may be not that clear, but it's not possible for user to input that data during normal use.